### PR TITLE
Add more options to export-vis command

### PR DESCRIPTION
**Pull request by clausnagel**
See #150

### CODE REVIEW

File: impexp-config/src/main/java/org/citydb/config/project/kmlExporter/GltfOptions.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    public GltfOptions() {
+        gltfVersion = GltfVersion.v2_0;
+        embedTextures = true;
+        useDracoCompression = true;
+
+        pathToConverter = "contribs" + File.separator + "collada2gltf";
+        String osName = System.getProperty("os.name").toLowerCase(Locale.ENGLISH);
+        if (osName.contains("windows"))
+            pathToConverter += File.separator + "COLLADA2GLTF-v2.1.3-windows-Release-x64" + File.separator + "COLLADA2GLTF-bin.exe";
```
Comment (BWibo) [inline impexp-config/src/main/java/org/citydb/config/project/kmlExporter/GltfOptions.java:61]: v 2.15 of COLLADA2glTF is available: https://github.com/KhronosGroup/COLLADA2GLTF/releases/tag/v2.1.5

**Review by BWibo — COMMENTED** [review]
For high res textures I sometimes use `textureImageScaleFactor` to down-scale texture images to increase visualization performance. 
`WallFillColor` and `RooFillfColor` are frequently used as well for styling or e.g. to create thematic visualizations in web client demos. It would be even cooler to be able to set these colors from a `GenericAttribute` or even map a `ColorPalette` to a value range. But this goes too far for this PR. 
Being able to set these values from the CLI and have them with the options below would already be great.

```
COLLADA/glTF rendering options:
  -s, --double-sided         Force all surfaces to be double sided.
      --no-surface-normals   Do not generate surface normals.
  -C, --crop-textures        Crop texture images.
  -x, --texture-atlas=<mode> Texture atlas mode: none, basic, tpim,
                               tpim_wo_rotation (default: basic).
      --no-pot-atlases       Do not create power-of-two sized texture atlase
```

I added some tests for the `AltitudeOptions` in https://github.com/tum-gis/3dcitydb-importer-exporter-docker/commit/f23a9824da77d13a60989a34443cb925c73e33e8 and it appears to be working fine. All other tests look good too.

File: impexp-config/src/main/java/org/citydb/config/project/kmlExporter/GltfOptions.java (hunk 1 above)
Comment (clausnagel) [reply impexp-config/src/main/java/org/citydb/config/project/kmlExporter/GltfOptions.java:61]: Yes, we have already tested with version 2.1.5 but it fails when using Draco compression. 2.1.4 seems to work though. We will have to do more tests to confirm.

**Comment by clausnagel** [comment]
Thanks for your continuous tests, @BWibo. This is very helpful.

I will add an option to set the image scale factor. Styling options are definetly out of scope of this PR though. You can define the styling in a config file and pass it via the `--config` option. It does not even have to be a full config but can be tailored to the styling content as shown below.

```xml
<kmlExport xmlns="http://www.3dcitydb.org/importer-exporter/config">
  <buildingStyles>
    <style type="collada">
    ...
    </style>
    ...
  </buildingStyles>
</kmlExport>
```

In general, I share your wish to be able to have more styling options like styling based on attributes and/or color ranges. I also think that we already should change the current styling options to be able to define colors for _all_ feature types. @Son-HNguyen, @thomashkolbe, maybe something you can take a look at?

**Comment by clausnagel** [comment]
Ok, I added the possibility that the `--bbox-tiling` can take a tile side length for automatic tiling (see #150).

I see this as last change to the `export-vis` command and want to merge this PR if there is no further feedback on the existing options.

File: impexp-config/src/main/java/org/citydb/config/project/kmlExporter/GltfOptions.java (hunk 1 above)
Comment (clausnagel) [reply impexp-config/src/main/java/org/citydb/config/project/kmlExporter/GltfOptions.java]: Updated COLLADA2glTF converter to v2.1.4 with 89a163d6326dc4d03e5cf6345b8e67f41338d60d. Further tests confirmed that version 2.1.5 has issues when using Draco compression.

**Comment by yaozhihang** [comment]
@clausnagel thanks for the nice work. The PR looks good to me. 
However, there is sitll an small issue regarding the query option. The current KML/Collada/glTF Exporter requires at least a gml:id filter or a bounding box filter. However, the CLI tells that the query option is optional. To solve this problem, we may need to implement the automatic calculation of bounding box. This should be done in a separate MR.

**Review by yaozhihang — APPROVED** [review]

**Comment by BWibo** [comment]
I added tests for `-f` and `-g auto=tileLength` in https://github.com/tum-gis/3dcitydb-importer-exporter-docker/commit/379ce1a34273f4216ab7dcccb0067216fda6a05a and everything seems fine.
Thx for the great work @clausnagel.

**Review by BWibo — APPROVED** [review]

**Comment by clausnagel** [comment]
@yaozhihang: Yes, `--bbox` and `--gml-id` are optional because the query settings can also be provided in a config file through the `--config` option. We still want to support the "old" way of using the CLI so that you can provide a "full" config file containing all export options. 

If you run an export without providing a bbox or gml:id list one or the other way, the export will fail with a corresponding error message.

I agree that the automatic calculation of the bbox is a much smarter solution, and we should address the topic soon.

**Pull request merged**